libsolidity/ast/Types.cpp

```diff
@@ -389,43 +541,100 @@ IntegerConstantType::IntegerConstantType(Literal const& _literal)
 	}
 }
 
-bool IntegerConstantType::isImplicitlyConvertibleTo(Type const& _convertTo) const
+bool ConstantNumberType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 {
-	if (auto targetType = dynamic_cast<IntegerType const*>(&_convertTo))
+	if (_convertTo.category() == Category::Integer)
 	{
+		auto targetType = dynamic_cast<IntegerType const*>(&_convertTo);
 		if (m_value == 0)
 			return true;
 		int forSignBit = (targetType->isSigned() ? 1 : 0);
-		if (m_value > 0)
+		if (isInteger())
 		{
-			if (m_value <= (u256(-1) >> (256 - targetType->numBits() + forSignBit)))
+			if (m_value > 0)
+			{
+				if (m_value <= (u256(-1) >> (256 - targetType->numBits() + forSignBit)))
+					return true;
+			}
+			else if (targetType->isSigned() && -m_value <= (u256(1) << (targetType->numBits() - forSignBit)))
 				return true;
+			return false;
 		}
-		else if (targetType->isSigned() && -m_value <= (u256(1) << (targetType->numBits() - forSignBit)))
+		else
+		{
+			if (m_value > 0)
+			{
+				if (leftOfRadix() <= (u256(-1) >> (256 - targetType->numBits() + forSignBit)))
+					return true;
+			}
+			else if (targetType->isSigned() && -leftOfRadix() <= (u256(1) << (targetType->numBits() - forSignBit)))
+				return true;
+			return false;
+		}
+	}
+	else if (_convertTo.category() == Category::Fixed)
+	{
+		auto targetType = dynamic_cast<FixedType const*>(&_convertTo);
+		if (m_value == 0)
 			return true;
-		return false;
+		int forSignBit = (targetType->isSigned() ? 1 : 0);
+		if (isInteger())
```

Comment:
This case is for if our current constantNumber is an integer or not, the else defaults to if it's a fixed type. If it's an integer, we only focus on the bits left of the radix. If it's a fixed type, we make sure that the size of the integer bits and the size of the fractional bits for the type we are converting to are greater or equal to what we currently have so that they fit, respectively. We have the sign bit for obvious reasons in there.